src/http_microhttpd.c

```diff
@@ -589,19 +609,26 @@ static int encode_and_redirect_to_splashpage(struct MHD_Connection *connection,
 		}
 	}
 
-	if (encoded[0]) {
+	if (config->fas_port) {
```

Comment:
Doesn't it makes sense to create the querystr here?
That is, place this block here:
```
	if (config->fas_secure_enabled != 1) {
		safe_asprintf(&querystr, "?clientip=%s&gatewayname=%s&tok=%s", client->ip, config->gw_name, client->token);
	} else {
		safe_asprintf(&querystr, "?clientip=%s&gatewayname=%s", client->ip, config->gw_name);
	}
```

Reply:
There is an argument for that but also there is a similar argument for merging together redirect_to_splashpage and encode_and_redirect_to_splashpage.
Lots of scope for refactoring the whole of NDS as some of the logic has become a little obscured.